src/logic/calculate.js

```diff
@@ -0,0 +1,16 @@
+import operate from './operate';
+
+const operations = ['+', '−', '×', '÷', '%', 'AC', '=', '+/-'];
+
+const calculate = (data, buttonName) => {
+  const { total, operation, next } = data;
+  let totalGen;
+  if (operations.includes(buttonName)) {
+    totalGen = operate(total, next, buttonName);
+    return { totalGen, next, buttonName };
```

Comment:
### Heads-Up:
**we need to add more cases;** 
1. if the `buttonName`  = `.` we should return `0.`